src/dune_rules/dune_file.ml

```diff
@@ -1686,7 +1706,34 @@ module Rule = struct
          field_o "alias"
            (Dune_lang.Syntax.since Stanza.syntax (2, 0) >>> Alias.Name.decode)
        in
-       { targets; deps; action; mode; locks; loc; enabled_if; alias; package })
+       let mode, patch_back_source_tree =
+         match mode with
+         | Normal mode -> (mode, false)
+         | Patch_back_source_tree ->
+           if
+             List.exists (Bindings.to_list deps) ~f:(function
+               | Dep_conf.Sandbox_config _ -> true
+               | _ -> false)
+           then
+             User_error.raise ~loc
+               [ Pp.text
+                   "Rules with (mode patch-back-source-tree) cannot have an \
+                    explicit sandbox configuration has it is implied by (mode \
+                    patch-back-source-tree)."
```

Comment:
... implied to be `(sandbox always)` to be more explicit?

Reply:
Technically it's `(sandbox copy)`, which is not expressible in a `dune` file. That's why I didn't put it